src/expandableCalendar/WeekCalendar/index.tsx

```diff
@@ -1,221 +1,225 @@
         context={currentContext}
-        numberOfDays={context.numberOfDays}
-        timelineLeftInset={context.timelineLeftInset}
+        onDayPress={onDayPressCallback}
+        numberOfDays={numberOfDays}
+        timelineLeftInset={timelineLeftInset}
       />
     );
-  };
+  },[firstDay, onDayPressCallback, context, date]);
 
-  getItemLayout = (_: any, index: number) => {
-    return {
-      length: this.containerWidth,
-      offset: this.containerWidth * index,
-      index
-    };
-  };
-
-  keyExtractor = (_: string, index: number) => index.toString();
+  const keyExtractor = useCallback((item) => item, []);
 
-  renderWeekDaysNames = () => {
+  const renderWeekDaysNames = useMemo(() => {
     return (
-      <WeekDaysNames 
-        firstDay={this.props.firstDay} 
-        style={this.style.dayHeader} 
+      <WeekDaysNames
+        firstDay={firstDay}
+        style={style.current.dayHeader}
       />
     );
-  };
+  },[firstDay]);
 
-  render() {
-    const {allowShadow, firstDay, hideDayNames, current, context} = this.props;
-    const {items} = this.state;
-    const extraData = Map({
-      current,
-      date: context.date,
-      firstDay
-    });
+  const weekCalendarStyle = useMemo(() => {
+    return [
+      allowShadow && style.current.containerShadow,
+      !hideDayNames && style.current.containerWrapper
+    ];
+  }, [allowShadow, hideDayNames]);
 
-    return (
-      <View
-        testID={this.props.testID}
-        style={[allowShadow && this.style.containerShadow, !hideDayNames && this.style.containerWrapper]}
-      >
-        {!hideDayNames && (
-          <View style={[this.style.week, this.style.weekCalendar]}>
-            {/* {this.props.weekNumbers && <Text allowFontScaling={false} style={this.style.dayHeader}></Text>} */}
-            {this.renderWeekDaysNames()}
-          </View>
-        )}
-        <FlatList
-          ref={this.presenter.list}
-          data={items}
-          extraData={extraData}
-          style={this.style.container}
-          horizontal
-          showsHorizontalScrollIndicator={false}
-          pagingEnabled
-          scrollEnabled
-          renderItem={this.renderItem}
-          keyExtractor={this.keyExtractor}
-          initialScrollIndex={NUMBER_OF_PAGES}
-          getItemLayout={this.getItemLayout}
-          onScroll={this.onScroll}
-          onMomentumScrollEnd={this.onMomentumScrollEnd}
-        />
+  const containerStyle = useMemo(() => {
+    return [style.current.week, style.current.weekCalendar];
+  }, []);
+
+  const flatListContainerStyle = useMemo(() => {
+    return style.current.container;
+  }, []);
+
+  const getItemLayout = useCallback((_, index: number) => {
+    return {
+      length: containerWidth,
+      offset: containerWidth * index,
+      index
+    };
+  }, [containerWidth]);
+
+  const onViewableItemsChanged = useCallback(({viewableItems}: { viewableItems: Array<ViewToken>}) => {
+    const newDate = viewableItems[0]?.item;
```

Comment:
You can move this line after the check since you don't need it in case you return here...